modules/dqt/jsx/react.filterBuilder.js

```diff
@@ -397,17 +399,26 @@ class FilterRule extends Component {
       );
     } else {
       // Else display dropdown for instrument select
-      let options = this.props.items.map((item, index) => {
-        return (
-          <option key={index} value={item.category}>{item.category}</option>
+      if (Array.isArray(this.props.items)) {
+        let options = this.props.items.map((item, index) => {
+          return (
+            <option key={index} value={item.category}>{item.category}</option>
+          );
+        });
+        rule = (
+          <select onChange={this.selectInstrument}
+                  className="input-sm col-xs-10">
+            <option value=''/>
+            {options}
+          </select>
         );
-      });
-      rule = (
-        <select onChange={this.selectInstrument} className="input-sm col-xs-10">
-          <option value=''/>
-          {options}
-        </select>
-      );
+      } else {
+        rule = (
+          <select className="input-sm col-xs-10">
```

Comment:
I don't understand why this is here. What is the point of a select element with one empty option, no name, no click handler, and no id?

Reply:
@driusan I think this addition was during the start of development for this PR and when I was working on having the DQT's UI function without any data from the backend. This code is now irrelevant because the functionality is hidden until the data is loaded.